x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/create_step_footer/create_step_footer.tsx

```diff
@@ -82,7 +82,12 @@ export const CreateStepFooter: FC<Props> = ({ jobId, jobType, showProgress }) =>
             jobStats.state === DATA_FRAME_TASK_STATE.STOPPED
           ) {
             clearInterval(interval);
-            setJobFinished(true);
+            // Check job has started. Jobs that fail to start will also have STOPPED state
+            setJobFinished(
```

Comment:
Is this check correct? Here I ran a classification job successfully and the link to view the results has never appeared:

![image](https://user-images.githubusercontent.com/7405507/94002122-5c5e7180-fd91-11ea-9f7d-46bc01935c88.png)

Reply:
@peteharverson - thanks for giving this a test! 🙏  I was only able to reproduce the issue mentioned above when the browser tab the job ran in was inactive. This is expected behavior as most browsers deprioritize inactive tabs and timers are heavily throttled. In this piece of code we are using `setInterval` to get the current job progress. 

https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API#Policies_in_place_to_aid_background_page_performance

https://stackoverflow.com/questions/57476352/interval-not-causing-re-render-in-react-component

Reply:
This is happening for me every time I run a job on the small `iris` data set, and making sure I don't leave the tab! Using this config:

```
{
  "id": "iris_classification4",
  "description": "",
  "source": {
    "index": [
      "iris"
    ],
    "query": {
      "match_all": {}
    }
  },
  "dest": {
    "index": "iris_classification4",
    "results_field": "ml"
  },
  "analysis": {
    "classification": {
      "dependent_variable": "class",
      "num_top_feature_importance_values": 5,
      "class_assignment_objective": "maximize_minimum_recall",
      "num_top_classes": 2,
      "prediction_field_name": "class_prediction",
      "training_percent": 12,
      "randomize_seed": 1462923820761784000
    }
  },
  "analyzed_fields": {
    "includes": [
      "class",
      "petal_length",
      "petal_width",
      "sepal_length",
      "sepal_width"
    ],
    "excludes": []
  },
  "model_memory_limit": "2mb",
  "create_time": 1600954687001,
  "version": "8.0.0",
  "allow_lazy_start": false,
  "max_num_threads": 1
}
```

Reply:
Added in some debug, and looks like when it fails to show the button the `jobStats.state` is left in the `analyzing` state, so `setJobFinished` must be getting passed `false`.

Reply:
Nice find, @peteharverson 👍  
Updated job finished check to make sure we're using up to date data. 7e290eefb2b313ea6eda44c72c6f0da9a78036c4
I confirmed I couldn't reproduce on my end anymore. Would you be up for confirming when you get a chance?

Reply:
Just tested it with the same job config can confirm I couldn't reproduce the issue any more

Reply:
I tried several tries and can no longer reproduce the issue I was seeing.